File: 1094. Car Pooling.cpp
class Solution {
public:
    bool carPooling(vector<vector<int>>& trips, int capacity) {
        vector<int> people(1005,0);
        bool flag=true;
        for(int i=0;i<trips.size();i++){
            for(int j=trips[i][1];j<trips[i][2];j++){
                people[j]+=trips[i][0];
                if(people[j]>capacity){
                    flag=false;
                    break;
                }
            }
        }
        return flag;
    }
};

File: 1094. Car Pooling.go
func carPooling(trips [][]int, capacity int) bool {
    var people [1005]int
    flag:=true

    for i := 0;i<len(trips);i++{
        for j := trips[i][1] ;j < trips[i][2];j++{
            people[j]+=trips[i][0]
            if(people[j]>capacity){
                flag=false
                break
            }
        }
    }
    return flag
}